code/modules/reagents/newchem/medicine.dm

```diff
@@ -0,0 +1,711 @@
+		var/obj/item/I = M.get_active_hand()
+		if(I)
+			M.drop_item()
+		M.adjustToxLoss(1*REM)
+		M.Dizzy(3)
+		M.Jitter(3)
+	..()
+	return
+datum/reagent/morphine/addiction_act_stage3(var/mob/living/M as mob)
+	if(prob(33))
+		var/obj/item/I = M.get_active_hand()
+		if(I)
+			M.drop_item()
+		M.adjustToxLoss(2*REM)
+		M.Dizzy(4)
+		M.Jitter(4)
+	..()
+	return
+datum/reagent/morphine/addiction_act_stage4(var/mob/living/M as mob)
+	if(prob(33))
+		var/obj/item/I = M.get_active_hand()
+		if(I)
+			M.drop_item()
+		M.adjustToxLoss(3*REM)
+		M.Dizzy(5)
+		M.Jitter(5)
+	..()
+	return
+
+datum/reagent/oculine/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	M.eye_blurry = max(M.eye_blurry-5 , 0)
+	M.eye_blind = max(M.eye_blind-5 , 0)
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		var/datum/organ/internal/eyes/E = H.internal_organs_by_name["eyes"]
+		if(istype(E))
+			if(E.damage > 0)
+				E.damage -= 1
+	..()
+	return
+/datum/chemical_reaction/oculine
+	name = "Oculine"
+	id = "oculine"
+	result = "oculine"
+	required_reagents = list("charcoal" = 1, "carbon" = 1, "hydrogen" = 1)
+	result_amount = 3
+	mix_message = "The mixture sputters loudly and becomes a pale pink color."
+
+datum/reagent/oculine
+	name = "Oculine"
+	id = "oculine"
+	description = "Cures blindness and heals eye damage over time."
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+	metabolization_rate = 0.4
+	var/cycle_amount = 0
+
+datum/reagent/atropine
+	name = "Atropine"
+	id = "atropine"
+	description = "If patients health is below -25 it will heal 3 brute and burn damage per cycle, as well as stop any oxyloss. Good for stabilising critical patients."
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+	metabolization_rate = 0.2
+	overdose_threshold = 35
+
+datum/reagent/atropine/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	if(M.health > -60)
+		M.adjustToxLoss(1*REM)
+	if(M.health < -25)
+		M.adjustBruteLoss(-3*REM)
+		M.adjustFireLoss(-3*REM)
+	if(M.oxyloss > 65)
+		M.setOxyLoss(65)
+	if(M.losebreath > 5)
+		M.losebreath = 5
+	if(prob(30))
+		M.Dizzy(5)
+		M.Jitter(5)
+	..()
+	return
+
+datum/reagent/atropine/overdose_process(var/mob/living/M as mob)
+	if(prob(50))
+		M.adjustToxLoss(2*REM)
+		M.Dizzy(1)
+		M.Jitter(1)
+	..()
+	return
+
+/datum/chemical_reaction/atropine
+	name = "Atropine"
+	id = "atropine"
+	result = "atropine"
+	required_reagents = list("ethanol" = 1, "acetone" = 1, "diethylamine" = 1, "phenol" = 1, "sacid" = 1)
+	result_amount = 5
+
+datum/reagent/epinephrine
+	name = "Epinephrine"
+	id = "epinephrine"
+	description = "Reduces most of the knockout/stun effects, minor stamina regeneration buff. Attempts to stop you taking too much oxygen damage. If the patient is in low to severe crit, heals toxins, brute, and burn very effectively. Will not heal patients who are almost dead. If overdosed will stun and deal toxin damage"
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+	metabolization_rate = 0.2
+	overdose_threshold = 30
```

Comment:
it's the goofchem stabilizer chem, and it only does that healing when you're deep in crit